olp-cpp-sdk-dataservice-read/src/repositories/NamedMutex.cpp

```diff
@@ -145,10 +145,15 @@ NamedMutex::NamedMutex(NamedMutexStorage& storage, const std::string& name,
 NamedMutex::~NamedMutex() { storage_.ReleaseLock(name_); }
 
 void NamedMutex::lock() {
-  if (!context_.IsCancelled() && !try_lock()) {
+  const bool valid = context_.ExecuteOrCancelled(
+      [&]() { return client::CancellationToken([&]() { Notify(); }); });
+
+  if (valid) {
     std::unique_lock<std::mutex> unique_lock{lock_mutex_};
     lock_condition_.wait(unique_lock,
                          [&] { return context_.IsCancelled() || try_lock(); });
+    // Reset the cancel token
+    context_.ExecuteOrCancelled([&]() { return client::CancellationToken(); });
```

Comment:
Why do you need to capture anything?
```suggestion
    context_.ExecuteOrCancelled([]() { return client::CancellationToken(); });
```

Reply:
As no side effect is expected, please consider this for the next PR